docs/blog/2019-10-14-adding-anchors-to-gatsby/index.md

```diff
@@ -0,0 +1,229 @@
+---
+title: "Adding Anchors 🔗 to Gatsby, using Sanity.io"
+date: 2019-10-14
+author: "ajonp"
+excerpt: "Have you ever hunted around for days trying to find that simple package for adding anchor links to your Gatsby blog? It is easier than you might think!"
+tags: ["browser-apis", "sanity"]
+canonicalLink: https://ajonp.com/blog/anchor-links-from-sanity-in-gatsby
+publishedAt: ajonp.com
+---
+
+![Gatsby Link](https://res.cloudinary.com/ajonp/image/upload/v1571083718/ajonp-ajonp-com/blog/gatsby-anchor-links/Gatsby_Sanity_Anchor_Links.webp)
+
+# Anchor Links 🔗 from Sanity in Gatsby
+
+> TL;DR version is make sure you implement `onRouteUpdate` and `shouldUpdateScroll` in `gatsby-browser.js`.
+
+## So what is an anchor link?
+
+Anchor links are a way to navigate within the same page in HTML. The easiest way to think of them are like a table of contents, or bookmarks on a page.
+You will see anchors used often in markdown pages that have header tags in the form of `#`. Now in order for those normal header tags to have a link they must be wrapped on the front end with a link tag, similar to this: `<a href="#anchor"><h2>Headline Link</h2></a>`. If you inspect the code on this page, you will even see an example of just that, as the blog is written in markdown and converted to HTML.
+
+## Sanity.io
+
+### How does this work with Sanity.io
+
+[Sanity](https://www.sanity.io/) is a headless content based CMS. You write in a [rich text editor](https://www.sanity.io/docs/what-you-need-to-know-about-block-text), which creates [portable text](https://www.portabletext.org/). So unlike markdown you wont have to convert header `#` items but you will have to serialize the portable text into something that Gatsby can understand. I won't dive too deeply into how you create a site using sanity.io there are some [great guides](https://www.gatsbyjs.org/packages/gatsby-source-sanity/?=sanity#gatsby-source-sanity) for that using `gatsby-source-sanity`.
+
+### Extending the Sanity Gatsby blog
+
+Sanity.io's Gatsby [blog example](https://www.sanity.io/guides/the-blog-template), is a great starting point for how to get up and running quickly. You can use that and then extend the functionality however you would like. In the example there is a file for posts which looks similar to below, what we really care about is the line `<div>{_rawBody && <PortableText blocks={_rawBody} />}</div>`.
+
+```tsx
```

Comment:
tsx ist not in the list in style guide - should we add it?


````suggestion
```jsx
````

Reply:
The filename is also visible in blog posts then we should add it also on all examples?
````suggestion
```tsx:title=my-filename.jsx
````